homeassistant/components/sensor/openhardwaremonitor.py

```diff
@@ -0,0 +1,195 @@
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+STATE_MIN_VALUE = 'minimal_value'
+STATE_MAX_VALUE = 'maximum_value'
+STATE_VALUE = 'value'
+STATE_OBJECT = 'object'
+CONF_INTERVAL = 'interval'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=1)
+SCAN_INTERVAL = timedelta(seconds=5)
+RETRY_INTERVAL = timedelta(seconds=30)
+
+OHM_VALUE = 'Value'
+OHM_MIN = 'Min'
+OHM_MAX = 'Max'
+OHM_CHILDREN = 'Children'
+OHM_NAME = 'Text'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=8085): cv.port
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Open Hardware Monitor platform."""
+    data = OpenHardwareMonitorData(config, hass)
+    add_devices(data.devices, True)
+
+
+class OpenHardwareMonitorDevice(Entity):
+    """Device used to display information from OpenHardwareMonitor."""
+
+    def __init__(self, data, name, path, unit_of_measurement):
+        """Initialize an OpenHardwareMonitor sensor."""
+        self._name = name
+        self._data = data
+        self.path = path
+        self.attributes = {}
+        self._unit_of_measurement = unit_of_measurement
+
+        self.value = None
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self.value
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the sun."""
+        return self.attributes
+
+    def update(self):
+        """Update the device from a new JSON object."""
+        self._data.update()
+        self._data.update_device(self)
+
+
+class OpenHardwareMonitorData(object):
+    """Class used to pull data from OHM and create sensors."""
+
+    def __init__(self, config, hass):
+        """Initialize the Open Hardware Monitor data-handler."""
+        self._data = None
+        self._config = config
+        self._hass = hass
+        self.devices = []
+        self.initialize(utcnow())
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Hit by the timer with the configured interval."""
+        if self._data is None:
+            self.initialize(utcnow())
+        else:
+            self.refresh()
+
+    def refresh(self):
+        """Download and parse JSON from OHM."""
+        data_url = "http://%s:%d/data.json" % (
+            self._config.get(CONF_HOST),
+            self._config.get(CONF_PORT))
+
+        try:
+            response = requests.get(data_url)
+            self._data = response.json()
+        except requests.exceptions.ConnectionError:
+            _LOGGER.error("ConnectionError: Is OpenHardwareMonitor running?")
+
+    def schedule_retry(self):
```

Comment:
This is to make sure the sensors are available, even if the API isn't available when starting hass. Until hass has the right way to actually setup a platform more than once (on error), this would be easier for the user.